rules/windows/network_connection/net_connection_win_notion_api_susp_communication.yml

```diff
@@ -0,0 +1,63 @@
+title: Potential Suspicious Network Connection To Notion API
+id: 7e9cf7b6-e827-11ed-a05b-15959c120003
+status: experimental
+description: Detects a non-browser process communicating with the Notion API. This could indicate potential use of a covert C2 channel such as "OffensiveNotion C2"
+references:
+    - https://github.com/mttaggart/OffensiveNotion
+    - https://medium.com/@huskyhacks.mk/we-put-a-c2-in-your-notetaking-app-offensivenotion-3e933bace332
+author: Gavin Knapp
+date: 2023/05/03
+tags:
+    - attack.command_and_control
+    - attack.t1102
+logsource:
+    product: windows
+    category: network_connection
+detection:
+    selection:
+        DestinationHostname|contains: 'api.notion.com'
+    filter_main_notion:
+        Image|endswith: '\AppData\Local\Programs\Notion\Notion.exe'
+    filter_optional_brave:
+        Image|endswith: '\brave.exe'
+    filter_optional_chrome:
+        Image:
+            - 'C:\Program Files\Google\Chrome\Application\chrome.exe'
+            - 'C:\Program Files (x86)\Google\Chrome\Application\chrome.exe'
+    filter_optional_firefox:
+        Image:
+            - 'C:\Program Files\Mozilla Firefox\firefox.exe'
+            - 'C:\Program Files (x86)\Mozilla Firefox\firefox.exe'
+    filter_optional_ie:
+        Image:
+            - 'C:\Program Files (x86)\Internet Explorer\iexplore.exe'
+            - 'C:\Program Files\Internet Explorer\iexplore.exe'
+    filter_optional_maxthon:
+        Image|endswith: '\maxthon.exe'
+    filter_optional_edge_1:
+        - Image|startswith: 'C:\Program Files (x86)\Microsoft\EdgeWebView\Application\'
+        - Image|endswith: '\WindowsApps\MicrosoftEdge.exe'
+        - Image:
+            - 'C:\Program Files (x86)\Microsoft\Edge\Application\msedge.exe'
+            - 'C:\Program Files\Microsoft\Edge\Application\msedge.exe'
+    filter_optional_edge_2:
+        Image|startswith:
+            - 'C:\Program Files (x86)\Microsoft\EdgeCore\'
+            - 'C:\Program Files\Microsoft\EdgeCore\'
+        Image|endswith:
+            - '\msedge.exe'
+            - '\msedgewebview2.exe'
+    filter_optional_opera:
+        Image|endswith: '\opera.exe'
+    filter_optional_safari:
+        Image|endswith: '\safari.exe'
+    filter_optional_seamonkey:
+        Image|endswith: '\seamonkey.exe'
+    filter_optional_vivaldi:
+        Image|endswith: '\vivaldi.exe'
+    filter_optional_whale:
+        Image|endswith: '\whale.exe'
+    condition: selection and not 1 of filter_main_* and not 1 of filter_optional_*
```

Comment:
going with the description, these filters should all be main filters, shouldn't they? @nasbench

Reply:
I agree on the `Image|endswith` which could be 1 selection. The image name itself should be enough to understand the selection

And good catch on the filter_optional_edge_1 selection. It probably needs to be split in 2 selections.

Reply:
@phantinuss  They should be main in this case yes. I'll change it.

@frack113 all new filters should be separated  by application or behaviour. Combined filters are not good for anyone as they cannot be removed easily by users and they loose the original meaning overtime as we have with so many broad filters :) 

The `filter_optional_edge_1` is already split because the `filter_optional_edge_2` needs an `and` or else it would be in the same selection.

Reply:
Also `C:\Program Files (x86)\Microsoft\EdgeWebView\Application\` can contain `msedgewebview2.exe` and `msedge.exe` that's why its there.

You can argue to split it into a third selection with an `and`

Reply:
As it is a low rule I can agree to have so many filter.
Now the condition make sense with only one "1 of filter".

Reply:
Basically all future rules will use either "optional" or "main" filters. 

Optional filters can be removed if you don't use the software for example discord or exchange, etc....

Main filters are part of the rule logic or system process that exists on every system

Reply:
@phantinuss merge after you approve.